tests/drivers/gpio/gpio_ite_it8xxx2_v2/src/main.c

```diff
@@ -0,0 +1,460 @@
+/*
+ * Copyright 2023 The ChromiumOS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/dt-bindings/gpio/ite-it8xxx2-gpio.h>
+#include <zephyr/fff.h>
+#include <zephyr/ztest.h>
+
+#define MY_GPIO DT_NODELABEL(gpioa)
+
+const struct device *const gpio_dev = DEVICE_DT_GET(MY_GPIO);
+static struct {
+	uint8_t fake;
+	uint8_t gpdmr;
+	uint8_t gpdr;
+	uint8_t gpotr;
+	uint8_t p18scr;
+	uint8_t wuemr, wuesr, wubemr;
+	uint8_t gpcr[DT_REG_SIZE_BY_IDX(MY_GPIO, 4)];
+	bool clear_gpcr_before_read;
+} registers;
```

Comment:
should there be a `__packed` here? Guess these happens to be aligned but it may not necessarily be the case

Reply:
Alignment/packing shouldn't matter - this doesn't have to match the actual hardware.  `fake_ecreg()` translates from the devicetree `reg` property to address of these struct members.